lib/src/model/broadcast/broadcast.dart

```diff
@@ -1,56 +1,47 @@
 import 'package:dartchess/dartchess.dart';
 import 'package:fast_immutable_collections/fast_immutable_collections.dart';
 import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:lichess_mobile/src/model/common/id.dart';
 
 part 'broadcast.freezed.dart';
 
+@freezed
+class BroadcastTournamentsListState with _$BroadcastTournamentsListState {
```

Comment:
I think we should start to use dart records instead of `@freezed` classes whenever possible. Also I would not use the `State` suffix because it is also the server response.

```dart
typedef BroadcastsList = ({
  IList<Broadcast> active,
  IList<Broadcast> upcoming,
  IList<Broadcast> past,
  int nextPage,
});
```

As long as there is no custom serialisation, or other logic, let's use a dart record. This can be done also for `BroadcastTournament`, `BroadcastRound`, `BroadcastGameSnapshot`, `BroadcastPlayer`.

Reply:
I can't use anymore `copyWith` with records. I use `copyWith` in the `BroadcastPaginator` to complete the broadcast list when I receive a new page.

Reply:
It's a tradeoff in terms of syntax and code readability. You can construct a new record and only update some fields:
```dart
state = ({
  updatedField: newValue,
  remains: record.remains,
})
```

In that case for instance it is perfectly readable.

I'd say in some cases `@freezed` will be more convenient thanks to the deep copy mechanism, so it is not a golden rule. But in general records will do fine I think.